python/rmm/_lib/memory_resource.pyx

```diff
@@ -268,8 +274,19 @@ cdef class LoggingResourceAdaptor(MemoryResource):
     cpdef get_file_name(self):
         return self._log_file_name
 
+
+class KeyInitializedDefaultDict(defaultdict):
```

Comment:
A short docstring describing the class would be helpful here

Reply:
docstring added